os/drivers/ai-soc/ndp120/src/ndp120_api.c

```diff
@@ -1012,29 +1037,33 @@ int ndp120_init(struct ndp120_dev_s *dev)
 	/* save handle so we can use it from debug routine later, e.g. from other util/shell */
 	_ndp_debug_handle = dev;
 
-	s = pthread_mutex_init(&dev->ndp_mutex_mbsync, NULL);
-	if (s) {
-		auddbg("failed to initialize mb sync mutex variable\n");
-	}
+	if (!reinit) {
 
-	s = pthread_mutex_init(&dev->ndp_mutex_mcu_mb_in, NULL);
-	if (s) {
-		auddbg("failed to initialize mb in mutex variable\n");
-	}
+		s = pthread_mutex_init(&dev->ndp_mutex_mbsync, NULL);
+		if (s) {
```

Comment:
Is it fine to proceed with the initialization if there is failure in the middle?

Reply:
This is not related to this PR. You are right, initialization should fail incase mutex_init fails. Let me add it in a new PR along with the comments from #6488 .